tests/cpp/test_gpu2.cpp

```diff
@@ -2670,57 +2687,23 @@ void testWelford(DataType dtype, int red_axis, int odim, int rdim) {
       "validate welford",
       reduction_params->lparams);
 }
-} // namespace
-
-TEST_F(NVFuserTest, FusionWelfordShmoo_CUDA) {
-  std::vector<DataType> dtypes = {
-      DataType::ComplexFloat,
-      DataType::ComplexDouble,
-      DataType::Double,
-      DataType::Float,
-      DataType::Half};
-
-  if (at::cuda::getDeviceProperties(0)->major >= 8) {
-    dtypes.insert(dtypes.end(), DataType::BFloat16);
-  }
-
-  std::vector<int> red_axis = {1, 0};
-  std::vector<int> output_dims = {160, 320};
-  std::vector<int64_t> red_dims;
-
-  // Tried to cut down the number iterations with just
-  // doing every other power of 2.
-  for (int i = 1; i <= 1024 * 1024; i <<= 2) {
-    red_dims.push_back(i);
-  }
-
-  for (auto dtype : dtypes) {
-    for (auto& axis : red_axis) {
-      for (auto& odim : output_dims) {
-        for (auto& rdim : red_dims) {
-          // TODO: original welford algorithm actually keeps a running sum of
-          // squares, i.e. M_{2n} in the
-          //       cf:
-          //       https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
-          //       algorithm notation, and it can reach inf for large numbers
-          //       with half precision. skipping too large volumes for half for
-          //       nwo might need further numerical experiments to re-design
-          //       this.
-          if (rdim > 32768 &&
-              (dtype == DataType::Half || dtype == DataType::BFloat16)) {
-            continue;
-          }
-          // Shmoo tests can occupy a lot of memory due to allocating many
-          // different tensor sizes. So in order to avoid an OOM during this
-          // test, we manually clear the allocator after it's reached a certain
-          // threshold.
-          maybeClearAllocator();
-          testWelford(dtype, axis, odim, rdim);
-        }
-      }
-    }
-  }
-}
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    WelfordReduction,
+    ::testing::Combine(
+        testing::ValuesIn(getFloatingDataTypes()), // data type
+        testing::ValuesIn(Pow2Vals1to1Million), // reduction dimension size
+        testing::Values(160, 320), // iteration dimension size
+        testing::Values(0, 1)), // reduction axis
+    [](const testing::TestParamInfo<WelfordReductionParams>& info)
+        -> std::string {
+      std::stringstream ss;
+      ss << "dtype_" << std::get<0>(info.param);
+      ss << "_rsize_" << std::get<1>(info.param);
+      ss << "_isize_" << std::get<2>(info.param);
+      ss << "_raxis_" << std::get<3>(info.param);
```

Comment:
```suggestion
      auto [dtype, rsize, isize, raxis] = info.param;
      ss << "dtype_" << dtype;
      ss << "_rsize_" << rsize;
      ss << "_isize_" << isize;
      ss << "_raxis_" << raxis;
```

Reply:
Also https://github.com/NVIDIA/Fuser/pull/2509/files#diff-9f148c6abf4059e23d7b0bb5f76736b320a711f920e899836e15cfa2e5ece398R2588 names these parameters slightly differently. It's better to make them consistent.

Reply:
revised.
To use structured bindings, I added `()` to enclouse labmda function `testing::TestParamInfo` to avoid compile errors. See https://github.com/google/googletest/issues/3848